dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventChangeLogService.java

```diff
@@ -48,13 +48,14 @@
 import org.hisp.dhis.dataelement.DataElement;
 import org.hisp.dhis.external.conf.DhisConfigurationProvider;
 import org.hisp.dhis.feedback.NotFoundException;
+import org.hisp.dhis.program.ChangeLogableEvent;
 import org.hisp.dhis.program.Event;
 import org.hisp.dhis.tracker.Page;
 import org.hisp.dhis.tracker.PageParams;
 import org.locationtech.jts.geom.Geometry;
 import org.springframework.transaction.annotation.Transactional;
 
-public abstract class EventChangeLogService<T> {
+public abstract class EventChangeLogService<T, R extends ChangeLogableEvent> {
```

Comment:
Minor thing, but by convention `R` should be used as the return type. I don't see it being returned from any method in this class, so I'm thinking if we should use `S` to indicate the second type, as it's recommended.